grammars/puppet.cson

```diff
@@ -394,23 +256,252 @@
       {
         'include': '#single-quoted-string'
       }
+      {
+        'include': '#bareword'
+      }
     ]
-  'variable':
+  'bareword':
+    'match': '(?<={)\\s*\\w+\\s*(?=})'
+    'name': 'constant.other.bareword.puppet'
+  'variables':
     'patterns': [
       {
         'captures':
           '1':
             'name': 'punctuation.definition.variable.puppet'
-        'match': '(\\$)([a-zA-Zx7f-xff\\$]|::)([a-zA-Z0-9_x7f-xff\\$]|::)*\\b'
+        'match': '(\\${*)([a-zA-Zx7f-xff\\$]|::)([a-zA-Z0-9_x7f-xff\\$]|::)*\\b'
         'name': 'variable.other.readwrite.global.puppet'
       }
       {
+        'match': '(\\$\\{)(?:[a-zA-Zx7f-xff\\$]|::)(?:[a-zA-Z0-9_x7f-xff\\$]|::)*(\\})'
```

Comment:
This match and the one on line 272 look fairly identical (other than some non-capturing groups and bracket matches) - can you explain the differences between the two?